ext/js/display/display-generator.js

```diff
@@ -356,6 +356,46 @@ export class DisplayGenerator {
         return node;
     }
 
+    /**
+     * @param {import('dictionary').InflectionHypothesis} inflectionHypothesis
+     * @returns {?HTMLElement}
+     */
+    _createInflectionHypothesis(inflectionHypothesis) {
+        const {source, inflections} = inflectionHypothesis;
+        if (!Array.isArray(inflections) || inflections.length === 0) { return null; }
+        const fragment = this._instantiate('inflection-hypothesis');
+
+        const sourceIcon = this._getInflectionSourceIcon(source);
+
+        fragment.appendChild(sourceIcon);
+
+        this._appendMultiple(fragment, this._createTermInflection.bind(this), inflections);
+        return fragment;
+    }
+
+    /**
+     * @param {import('dictionary').InflectionSource} source
+     * @returns {HTMLElement}
+     */
+    _getInflectionSourceIcon(source) {
+        const icon = document.createElement('span');
+        icon.style.marginRight = '0.5em';
+        switch (source) {
+            case 'dictionary':
```

Comment:
Just a suggestion, but is there any way the user can change the icons?

Reply:
Should be doable with custom css. Would also be nice to have actual icons.

Reply:
Going to revive this; can you make it so that icon has no `textContent` and instead has a `dataset.` property which specifies the type, and the content is driven by a CSS `::after` property? This will make it easier to style and also make it so the icon isn't selectable.